### PR TITLE
Fix issue where 'remove parens' would break code with ++/-- operators.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/29454

Also fixes a related issue i found while looking into this area where we could break code of the form `x+(++y)`.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jcouv @dotnet/roslyn-ide @jasonmalinowski

**Comment by CyrusNajmabadi** [comment]
Tagging @jcouv @dotnet/roslyn-ide @jasonmalinowski

File: src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs [hunk 1]
```diff
@@ -24,6 +25,19 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node,
                 ? node.Parent.Parent as ExpressionSyntax
                 : node.Parent as ExpressionSyntax;
 
+            // Have to be careful if we would remove parens and cause a + and a + to become a ++.
+            // (same with - as well).
+            var tokenBeforeParen = node.GetFirstToken().GetPreviousToken();
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs]: Does this have problems in script/interactive mode?

Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs]: Fortunately no.  `GetPreviousToken` will simply return `default(SyntaxToken)`.  `default(SyntaxToken)` returns empty-string for `.Text`, and `LastOrDefault` will just return `0` in that case.

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**